cmd/root.go

```diff
@@ -357,22 +361,44 @@ func updateIssue(ghIssue github.Issue, jIssue jira.Issue, ghClient github.Client
 		fields.Unknowns[key] = time.Now().Format(dateFormat)
 
 		fields.Type = jIssue.Fields.Type
-		fields.Summary = jIssue.Fields.Summary
+		if fields.Summary == "" {
+			fields.Summary = jIssue.Fields.Summary
+		}
 
 		issue := &jira.Issue{
 			Fields: &fields,
 			Key:    jIssue.Key,
 			ID:     jIssue.ID,
 		}
 
-		issue, res, err := jClient.Issue.Update(issue)
+		if !dryRun {
```

Comment:
this will not scale well and will get unmanageable quite fast. the behavior of this function is now being drastically altered by a variable that is not a param. Testing this is messy and ensuring correctness becomes increasingly difficult to reason about. The correct way to do this is for all of these functions to be methods on a issue-sync config struct that encapsulates all the configuration, including dry-run. These methods then operate on a method receiver that holds all the information needed to control the operation of the function. Either that, or all of these methods take an additional parameter `dryRun`.